core/src/main/java/fi/iki/express/NanoExpress.java

```
@@ -0,0 +1,180 @@
+     *
+     * Implementor of Router Interface will have full control of what will be served out from NanoHTTPD.
+     *
+     * <p/>
+     * (By default, this returns a 404 "Not Found" plain text error response.)
+     *
+     * @param session The HTTP session
+     * @return HTTP response, see class Response for details
+     */
+    @Override
+    public NanoHTTPD.Response serve(IHTTPSession session) {
+        Method method = session.getMethod();
+        NanoHTTPD.Response res = null;
+        try {
+            for (String uri : route_priority) {
+                Router r = routerArray.get(uri);
+                //This will follow the methods defined in NanoHTTPD.Method
+                switch (method) {
+                    case GET:
+                        res = r.doGet(session);
+                        break;
+                    case PUT:
+                        res = r.doPut(session);
+                        break;
+                    case POST:
+                        res = r.doPost(session);
+                        break;
+                    case DELETE:
+                        res = r.doDelete(session);
+                        break;
+                    case HEAD:
+                        res = r.doHead(session);
+                        break;
+                    case OPTIONS:
+                        res = r.doOptions(session);
+                        break;
+                    case TRACE:
+                        res = r.doTrace(session);
+                        break;
+                    case CONNECT:
+                        res = r.doConnect(session);
+                        break;
+                    case PATCH:
+                        res = r.doPatch(session);
+                        break;
+                    default:
+                        res = new IStatusResponse(Response.Status.METHOD_NOT_ALLOWED);
+                }
+                if (res != null) return res;
+            }
+        } catch (Exception e) {
```

From the engine perspective, it is to preserved the ability to continue to serving. So having the catch all exception here does just that. You still can do a stack trace here. But it is the responsibility of those who implement each Route class to perform the error handling.